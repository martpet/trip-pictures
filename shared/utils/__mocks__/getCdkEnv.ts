import { EnvName } from '../../types/EnvName';

const envName: EnvName = 'production';

export const getCdkEnv = vi.fn().mockName('getCdkEnv').mockReturnValue(envName);
