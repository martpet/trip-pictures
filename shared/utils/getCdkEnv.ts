import { envNames } from '../consts/envNames';
import { EnvName } from '../types/EnvName';

export function getCdkEnv() {
  const envName = process.env.ENV_NAME as EnvName;
  const isValid = envNames.includes(envName);

  if (!isValid) {
    throw Error('"process.env.ENV_NAME" is invalid');
  }

  return envName;
}
