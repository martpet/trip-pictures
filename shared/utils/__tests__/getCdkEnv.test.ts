import { getCdkEnv } from '../getCdkEnv';

beforeEach(() => {
  process.env.ENV_NAME = 'production';
});

describe('getCdkEnv', () => {
  it('returns a correct value', () => {
    expect(getCdkEnv()).toMatchSnapshot();
  });

  describe('when envName is invalid', () => {
    beforeEach(() => {
      process.env.ENV_NAME = 'foobar';
    });

    it('throws an error', () => {
      expect(() => {
        getCdkEnv();
      }).toThrowErrorMatchingSnapshot();
    });
  });
});
