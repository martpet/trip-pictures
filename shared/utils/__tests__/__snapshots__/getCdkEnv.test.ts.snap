// Vitest Snapshot v1

exports[`getCdkEnv > returns a correct value 1`] = `"production"`;

exports[`getCdkEnv > when envName is invalid > throws an error 1`] = `"\\"process.env.ENV_NAME\\" is invalid"`;
